lib/Extension/Prophecy/stubs/Prophecy.stub

```diff
@@ -24,6 +24,42 @@ class ObjectProphecy
     public function reveal()
     {
     }
+
+    /**
+     * Forces double to extend specific class.
+     *
+     * @template U
+     * @param class-string<U> $class
+     *
+     * @return self|T|U
```

Comment:
should be `ObjectProphecy<T>` ?

Reply:
Well it should be `ObjectProphecy<T&U>` but as far as I know, we don't support that.

Reply:
Arguably that might be what the issue is about :)

Reply:
And it is "T" as it extends, it is not an intersection

Reply:
Yeah I thought so, then I might need to dig into the code a little more.